### ISSUE TITLE
Main file missing from sketch

### ISSUE THREAD

**Issue by Lisoveliy** [microsoft/vscode-arduino#1665]
Hi. I think, it's a bug.  After initializing arduino "project" in arduino.json i have that data. 
![image](https://github.com/microsoft/vscode-arduino/assets/56991906/6782f20a-3d00-45d0-9447-f9cab08b409b)
File main.ino exists
![image](https://github.com/microsoft/vscode-arduino/assets/56991906/acc76851-f3ed-43f2-be12-ea592f531825)
But after try to compile i have that error.
![image](https://github.com/microsoft/vscode-arduino/assets/56991906/ad783c8e-f8ff-4b04-9c63-f468a7ee8ad7)
Entry point set in arduino.json as "main.ino" but extension also trying to find sketch named like folder. Please fix that, or maybe i mistaked somewhere?

**Comment by GimpMaster** [comment]
I'm getting the same issue as well. Any suggestions?

**Comment by GimpMaster** [comment]
It seems like if you rename your sketch to match the folder name it works, Seems like a bug.

**Comment by uztbt** [comment]
Is there any configuration parameter to specify a custom main file name?

**Comment by Lisoveliy** [comment]
I think, it is. It named "sketch" but it doesn't working.

**Comment by jdsalmonson** [comment]
Having this issue as well.  GimpMaster's solution works for me, but have we determined if this is a bug?

**Comment by spinpwr** [comment]
Using the same name for the folder (i.e. "project") and the main file is the same what the Arduino IDE does, right?
That could be a reason, but still we have a config for the main "sketch" which doesn't work.. so it's a bug for me

**Comment by spinpwr** [comment]
Using the same name for the folder (i.e. "project") and the main file is the same what the Arduino IDE does, right?
That could be a reason, but still we have a config for the main "sketch" which doesn't work.. so it's a bug for me

**Comment by OriginalCREart** [comment]
> Using the same name for the folder (i.e. "project") and the main file is the same what the Arduino IDE does, right? That could be a reason, but still we have a config for the main "sketch" which doesn't work.. so it's a bug for me

It turns out that the main sketch **MUST** be the same name of the container folder, I was having a file called **WifiManager.ino** inside a folder called **WifiManager** [**WifiManager\WifiManager.ino**], I renamed the **WifiManager.ino** to **Ultrasonic.ino**, and the verification/compilation/upload was failing, after reading @spinpwr comment, I renamed the folder to be as the same as the new sketch name **Ultrasonic.ino** but without the .ino extension and I ended up having [**Ultrasonic\Ultrasonic.ino**] and everything worked just fine and I was able to verify/compile/upload and rebuild IntelliSense configuration [_shortcut -> "Ctrl+Alt+I"_].

Thanks

**Comment by HeatPhoenix** [comment]
Agreed, the .json file should determine the main file. Not the folder name.

**Comment by manaspace** [comment]
Do not use Arduino Cli. Use Arduino Path and command targeting to your arduino ide folder, solves the error.

**Comment by alelavoie** [comment]
Looks like a bug indeed. To fix the issue, I simply created an empty `.ino` file with the same name as the project directory in the root of the project: `My Project/My Project.ino`

**Comment by QuentinHsu** [comment]
> Agreed, the .json file should determine the main file. Not the folder name.

I agree too! Otherwise, what's the point of the `sketch` field in the generated `.vscode/arduino.json`?!

**Comment by msingh0101** [comment]
so is this gonna get marked as a bug or not? 
Also, why cant the extension detect the working file rather than inferring it or using arduino.json (when fixed)?

**Comment by vadixidav** [comment]
Still an issue.

**Comment by Facul-101** [comment]
When will it be fixed?
